types/module/server/msgs.go

```diff
@@ -0,0 +1,88 @@
+package server
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// isServiceMsg checks if a type URL corresponds to a service method name,
+// i.e. /cosmos.bank.Msg/Send vs /cosmos.bank.MsgSend.
+// Function copied from cosmos-sdk, once we add ADR 033 and group module to it,
+// we'll want to merge them.
+func isServiceMsg(typeURL string) bool {
```

Comment:
Do u think it is sufficient to count number of slashes here? What If I put `http://`, will it be the the valid typeURL?

Reply:
This is directly pulled from ADR 031 implementation. `Any` type URLs won't take this form (https://developers.google.com/protocol-buffers/docs/proto3#any) and in the particular context of the sdk for `Msg`s, we do know that they can only look either like  `/cosmos.bank.Msg/Send` or `/cosmos.bank.MsgSend` for instance.